src/main/java/org/icgc/argo/program_service/controller/ProgramController.java

```diff
@@ -0,0 +1,119 @@
+  @Autowired private Grpc2JsonConverter grpc2JsonConverter;
+  @Autowired private AuthorizationService authorizationService;
+
+  @PostMapping(value = "/createProgram")
+  public ResponseEntity<CreateProgramResponseDTO> createProgram(
+      @RequestBody CreateProgramRequestDTO createProgramRequestDTO) throws IOException {
+    authorizationService.requireDCCAdmin();
+    CreateProgramRequest request =
+        grpc2JsonConverter.fromJson(
+            grpc2JsonConverter.getJsonFromObject(createProgramRequestDTO),
+            CreateProgramRequest.class);
+    CreateProgramResponse response = serviceFacade.createProgram(request);
+    return new ResponseEntity(
+        grpc2JsonConverter.prepareCreateProgramResponse(response), HttpStatus.CREATED);
+  }
+
+  @DeleteMapping(value = "/removeProgram")
+  public void removeProgram(@RequestBody RemoveProgramRequestDTO removeProgramRequestDTO) {
+    authorizationService.requireDCCAdmin();
+    RemoveProgramRequest request;
+    try {
+      request =
+          grpc2JsonConverter.fromJson(
+              grpc2JsonConverter.getJsonFromObject(removeProgramRequestDTO),
+              RemoveProgramRequest.class);
+      serviceFacade.removeProgram(request);
+    } catch (EmptyResultDataAccessException | InvalidDataAccessApiUsageException | IOException e) {
+      log.error("Exception throw in removeProgram: {}", e.getMessage());
+      throw new NotFoundException(ExceptionUtils.getStackTrace(e));
+    }
+  }
+
+  @PutMapping(value = "/updateProgram")
+  public ResponseEntity<UpdateProgramResponseDTO> updateProgram(
+      @RequestBody UpdateProgramRequestDTO updateProgramRequestDTO) {
+    authorizationService.requireDCCAdmin();
+    UpdateProgramRequest request;
+    UpdateProgramResponse response;
+    try {
+      request =
+          grpc2JsonConverter.fromJson(
+              grpc2JsonConverter.getJsonFromObject(updateProgramRequestDTO),
+              UpdateProgramRequest.class);
+      response = serviceFacade.updateProgram(request);
+    } catch (NotFoundException | NoSuchElementException | IOException e) {
+      log.error("Exception throw in updateProgram: {}", e.getMessage());
+      throw new NotFoundException(ExceptionUtils.getStackTrace(e));
+    }
+    return new ResponseEntity(
+        grpc2JsonConverter.prepareUpdateProgramResponse(response), HttpStatus.OK);
+  }
+
+  @GetMapping(value = "/getProgram/{shortName}")
+  public ResponseEntity<GetProgramResponseDTO> getProgram(
+      @PathVariable(value = "shortName", required = true) String shortName) throws IOException {
+    authorizationService.requireProgramUser(shortName);
+    GetProgramRequest request =
+        GetProgramRequest.newBuilder().setShortName(StringValue.of(shortName)).build();
+    GetProgramResponse response = serviceFacade.getProgram(request);
+    return new ResponseEntity(
+        grpc2JsonConverter.prepareGetProgramResponse(response), HttpStatus.OK);
+  }
+
+  @PostMapping(value = "/activateProgram")

```

Comment:
This one does need an additional path, can we just use `/activate`?

POST `/programs/activate`

Reply:
Modified request mapping to /activate